### PR TITLE
Multiple list view crudely working

### LINKED ISSUE #29: Play Next and Suggestions in Queue
The Queue UI should support showing Play Next and Suggestions in one view with a separator between them.

### PR DESCRIPTION
**Pull request by daniel-centore**
Lots of stuff still doesn't work but you can see the lists of songs. Checkout line 60 of QueueView for where you're going to be adding in play next support. I'm going to keep working to add in support for all the other features we need in the queue. Just wanted to unblock you.

Fixes #29 
Progress on #85 

simulator screen shot apr 11 2017 8 52 12 pm

### CODE REVIEW

File: MeNext/Layout/ResultList/ListsView.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        /// Initializes a new instance of the <see cref="T:MeNext.QueuesView`1"/> class. This constructor is for
+        /// multiple lists of elements, represented by GroupWrappers.
+        /// </summary>
+        /// <param name="groups">The list of GroupWrappers</param>
+        public ListsView(params ResultsGroup<T>[] groups)
+        {
+            this.HasUnevenRows = true;
+
+            this.IsGroupingEnabled = true;
+            this.GroupDisplayBinding = new Binding("Title");  // TODO
```
Comment (ConnorFoody) [inline MeNext/Layout/ResultList/ListsView.cs:36]: Is this too fragile? Should we make a constant for Title? Is there a good way to get around using bindings?

File: MeNext/Layout/ResultList/ListsView.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            //});
+        }
+
+        public void UpdateLists(params IEnumerable<T>[] itemses)
+        {
+            for (int i = 0; i < itemses.Length; ++i) {
+                var items = itemses[i];
+                var group = this.allGroups[i];
+                var wrappedItems = new List<ResultItemData>();
+                foreach (var item in items) {
```
Comment (ConnorFoody) [inline MeNext/Layout/ResultList/ListsView.cs:60]: Maybe add some comments to this code? Not sure clear what's happening here.

File: MeNext/Layout/ResultList/ListsView.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            //{
+            //    return ell();
+            //});
+        }
+
+        public void UpdateLists(params IEnumerable<T>[] itemses)
+        {
+            for (int i = 0; i < itemses.Length; ++i) {
+                var items = itemses[i];
+                var group = this.allGroups[i];
```
Comment (ConnorFoody) [inline MeNext/Layout/ResultList/ListsView.cs:58]: this seems a little dangerous? Maybe make sure the lists are the same length?

File: MeNext/Layout/ResultList/ResultListCell.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                RowDefinitions = rows,
+            };
+            grid.Children.Add(metadata, 0, 0);
+            grid.Children.Add(buttons, 1, 0);
+
+            this.View = grid;
+        }
+
+        // This is called when the backing data for the cell changes
+        protected override void OnBindingContextChanged()
```
Comment (ConnorFoody) [inline MeNext/Layout/ResultList/ResultListCell.cs:116]: Like when? Don't we just blow everything away when we get new data?

File: MeNext/Layout/ResultList/ResultListCell.cs [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            this.subtitleLabel = new Label
+            {
+                LineBreakMode = LineBreakMode.TailTruncation
+            };
+
+            this.taskButton = new Button
+            {
+                Text = SUGGESTIONS_ADD,
+                WidthRequest = BUTTON_WIDTH,
+                Command = new Command((obj) =>
```
Comment (ConnorFoody) [inline MeNext/Layout/ResultList/ResultListCell.cs:46]: I assume it will be possible to give it different cmds?

**Review by ConnorFoody — APPROVED** [review]
Generally LGTM. Just a few little things. 

Also, unless you plan to use it maybe remove commented out code?

File: MeNext/Layout/ResultList/ListsView.cs (hunk 1 above)
Comment (daniel-centore) [reply MeNext/Layout/ResultList/ListsView.cs:36]: Yeah, I hate bindings and avoided their use as much as possible. That TODO with no helpful information is actually for me to look into using reflection witchcraft to be less fragile.

File: MeNext/Layout/ResultList/ListsView.cs (hunk 3 above)
Comment (daniel-centore) [reply MeNext/Layout/ResultList/ListsView.cs:58]: I'll add an assert, but my local branch has progressed significantly so it'll be in the next PR.

File: MeNext/Layout/ResultList/ListsView.cs (hunk 2 above)
Comment (daniel-centore) [reply MeNext/Layout/ResultList/ListsView.cs:60]: Done; Next PR

File: MeNext/Layout/ResultList/ResultListCell.cs (hunk 5 above)
Comment (daniel-centore) [reply MeNext/Layout/ResultList/ResultListCell.cs:46]: Yes; in progress.

File: MeNext/Layout/ResultList/ResultListCell.cs (hunk 4 above)
Comment (daniel-centore) [reply MeNext/Layout/ResultList/ResultListCell.cs:116]: No. We reuse all the UI elements for efficiency.

**Review by daniel-centore — COMMENTED** [review]
Most of the commented sections are going to be used elsewhere, still work in progress. Just wanted to get this in to unblock play next queue work.

**Pull request merged**